File: 14.二分查找/Solution.py
        i, j = 0, len(nums)-1
        while i < j:
            mid = (i + j)//2
            if nums[mid] > target:
                j = mid - 1
            elif nums[mid] < target:
                i = mid + 1
            elif nums[mid] == target:
                j = mid
        if nums[j] == target:
            return j
        return -1

File: 14.二分查找/Solution.cpp
class Solution {
public:
    /**
     * @param nums: The integer array.
     * @param target: Target number to find.
     * @return: The first position of target. Position starts from 0. 
     */
    int binarySearch(vector<int> &array, int target) {
        // write your code here
        long long r = array.size() - 1;
        long long l = 0;
        long long mid;
        while(l < r){
            mid = (l + r)/2;
            if(array[mid] == target)
                r = mid;
            else if(array[mid] < target)
                l = mid + 1;
            else
                r = mid - 1;
        }
        if(array[r] == target) return r;
        return -1;
    }
};